.github/tools-cache/action.yaml

```diff
@@ -0,0 +1,21 @@
+---
+name: tools-cache
+description: Caches development tools
+runs:
+  using: composite
+  steps:
+    - uses: actions/cache@v3
+      id: tools-cache
+      with:
+        path: ./tmp/bin
+        key: ${{ runner.os }}-tools-${{ hashFiles('./.github/tools-cache') }}
```

Comment:
We need to may be hash `hack/tools.sh` instead. What ObO does is to create a '.github/tools` file see: https://github.com/rhobs/observability-operator/blob/main/.github/tools
The hash of that file is used to check of new binaries need to be downloaded.